apps/condo/domains/miniapp/schema/B2CAppProperty.js

```diff
@@ -37,37 +37,43 @@ const B2CAppProperty = new GQLListSchema('B2CAppProperty', {
-            type: Text,
-            isRequired: true,
-            hooks: {
-                resolveInput: ({ resolvedData, fieldPath }) => {
-                    if (typeof resolvedData[fieldPath] === 'string') {
-                        resolvedData[fieldPath] = resolvedData[fieldPath].toLowerCase()
-                    }
-                    return resolvedData[fieldPath]
-                },
-                validateInput: async ({ resolvedData, fieldPath, addFieldValidationError }) => {
-                    const inputAddress = resolvedData[fieldPath]
-                    if (!inputAddress) {
-                        return addFieldValidationError(INCORRECT_ADDRESS_ERROR)
-                    }
-                    // TODO(DOMA-3250) Fix this mess after migration to new address-suggestion service
-                    const suggestions = await getAddressSuggestions(inputAddress, 1)
-                    const suggestionAddress = get(suggestions, ['0', 'value'], '')
-                    const suggestionHouseType = get(suggestions, ['0', 'data', 'house_type_full'])
-                    if (!VALID_HOUSE_TYPES.includes(suggestionHouseType)) {
-                        return addFieldValidationError(`${INCORRECT_HOUSE_TYPE_ERROR}. Valid values are: [${VALID_HOUSE_TYPES.join(', ')}]`)
-                    }
-                    if (suggestionAddress.toLowerCase() !== inputAddress.toLowerCase()) {
-                        return addFieldValidationError(INCORRECT_ADDRESS_ERROR)
-                    }
+    },
+    plugins: [
+        uuided(),
+        addressService({
+            fieldsHooks: {
+                address: {
+                    resolveInput: ({ resolvedData, fieldPath }) => {
+                        if (typeof resolvedData[fieldPath] === 'string') {
+                            resolvedData[fieldPath] = resolvedData[fieldPath].toLowerCase()
+                        }
+                        return resolvedData[fieldPath]
+                    },
+                    validateInput: ({ resolvedData, fieldPath, addFieldValidationError, operation }) => {
+                        const inputAddress = resolvedData[fieldPath]
+                        if (!inputAddress) {
+                            return addFieldValidationError(INCORRECT_ADDRESS_ERROR)
+                        }
+
+                        const suggestionAddress = get(resolvedData, ['addressMeta', 'value'], '')
+                        const suggestionHouseType = get(resolvedData, ['addressMeta', 'data', 'house_type_full'])
+
+                        if (!VALID_HOUSE_TYPES.includes(suggestionHouseType)) {
+                            return addFieldValidationError(`${INCORRECT_HOUSE_TYPE_ERROR}. Valid values are: [${VALID_HOUSE_TYPES.join(', ')}]`)
+                        }
+                        if (operation === 'create' && suggestionAddress.toLowerCase() !== inputAddress.toLowerCase()) {
```

Comment:
Why did this check become create-only?

Reply:
Agree with the question asked.
What if we change the address?